CodeEdit/Features/SourceControl/Views/SourceControlPullView.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  SourceControlPullView.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 6/28/24.
+//
+
+import SwiftUI
+
+struct SourceControlPullView: View {
+    @Environment(\.dismiss)
+    private var dismiss
+
+    @EnvironmentObject var sourceControlManager: SourceControlManager
+
+    @State var loading: Bool = false
+
+    var body: some View {
+        VStack(spacing: 0) {
+            Form {
+                Section {
+                    RemoteBranchPicker(
+                        branch: $sourceControlManager.operationBranch,
+                        remote: $sourceControlManager.operationRemote,
+                        onSubmit: submit,
+                        canCreateBranch: false
+                    )
+                } header: {
+                    Text("Pull remote changes from")
+                }
+                Section {
+                    Toggle("Rebase local changes onto upstream changes", isOn: $sourceControlManager.operationRebase)
+                }
+            }
+            .formStyle(.grouped)
+            .scrollDisabled(true)
+            .scrollContentBackground(.hidden)
+            HStack {
+                if loading {
+                    HStack(spacing: 7.5) {
+                        ProgressView()
+                            .progressViewStyle(.circular)
+                            .controlSize(.small)
+                        Text("Pulling changes...")
+                            .font(.subheadline)
+                    }
+                }
+                Spacer()
+                Button {
+                    dismiss()
+                } label: {
+                    Text("Cancel")
+                        .frame(minWidth: 56)
+                }
+                .disabled(loading)
+                Button(action: submit) {
+                    Text("Pull")
+                        .frame(minWidth: 56)
+                }
+                .buttonStyle(.borderedProminent)
+                .disabled(loading)
+            }
+            .padding(.horizontal, 20)
+            .padding(.bottom, 20)
+        }
+        .frame(minWidth: 500)
+    }
+
+    func submit() {
```

Comment:
Could you add documentation or change the function name? The function in SourceControlPushView has the same name and looks similar, which could lead to confusion

Reply:
You are submitting the form in both files. I thought that was clear enough. If the function were global I might understand.

Reply:
Makes sense. Please add some documentation (a really short note is enough), like: ‘Pulls changes, shows stash sheet if needed

Reply:
Done